### ISSUE TITLE
Linter fails (or is very slow) on large configuration files.

### ISSUE THREAD

**Issue by adam-hulin** [jhinch/nginx-linter#33]
I have a generated nginx configuration file that is 3.7MB / 130K LOC.
Trying to run nginx-linter on that file fails;

```
nginx-linter --include nginx.conf 

nginx.conf
Unexpected error: RangeError: Maximum call stack size exceeded
    at flatten (/opt/homebrew/lib/node_modules/nginx-linter/node_modules/table/dist/src/utils.js:112:15)
    at mapDataUsingRowHeights (/opt/homebrew/lib/node_modules/nginx-linter/node_modules/table/dist/src/mapDataUsingRowHeights.js:49:32)
    at table (/opt/homebrew/lib/node_modules/nginx-linter/node_modules/table/dist/src/table.js:24:64)
    at outputInnerResults (/opt/homebrew/lib/node_modules/nginx-linter/bin/_cli/commands.js:148:20)
    at outputResults (/opt/homebrew/lib/node_modules/nginx-linter/bin/_cli/commands.js:127:5)
    at /opt/homebrew/lib/node_modules/nginx-linter/bin/_cli/commands.js:83:13
    at Array.map (<anonymous>)
    at validate (/opt/homebrew/lib/node_modules/nginx-linter/bin/_cli/commands.js:80:32)
    at execute (/opt/homebrew/lib/node_modules/nginx-linter/bin/_cli/commands.js:51:20)
    at Object.main (/opt/homebrew/lib/node_modules/nginx-linter/bin/_cli/commands.js:168:16)
```

I also have a generated configuration file that is 800KB / 27500 LOC.
Running the linter on this file takes about 30 minutes on an Apple M1 CPU.
The file has a lot of whitespace issues, if that's a hint.

**Comment by jhinch** [comment]
Hi, thanks for raising the bug. I had a few questions:

* The error you are getting seems to indicate recursion. The function is getting the error on is a recursive function which recurses for each level of included file. Does this large file also have a deep number of includes? I can likely change the function to be iterative instead of recursive but im not sure that will help with the performance issues. 
* Is there any other hints you can give in terms of helping me reproduce the issue. For example is the file a deep number of nested location blocks? is it reasonably flat in form etc?

**Comment by jhinch** [comment]
I did some testing locally. It seems like this is related to the dependency `table` I'm using to format the error results in a column format. I think its not about the size of the file but the number of errors in any one file. When I profile a large 100K LOC file locally, the linting happens within a second or two but the rendering of the results takes 30 seconds. This appears to be a known issue with the dependency: https://github.com/gajus/table/issues/223

**Comment by jhinch** [comment]
At least in the local testing I did a few months ago, I don't think it fixes the problem.

**Comment by jhinch** [comment]
I have put in a fix for this in 0.7.0

The fix was to limit the number of errors to 1000 per lint type. For my test of a large file it went from 10 seconds to validate to 2 seconds
